src/shared/driverUtils.js

```diff
@@ -0,0 +1,258 @@
+/*
+ * Copyright 2021-2022 Inclusive Design Research Centre, OCAD University
+ * All rights reserved.
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ */
+
+"use strict";
+
+const fluid = require("infusion");
+const axios = require("axios");
+const { exec, execSync } = require("child_process");
+
+fluid.registerNamespace("fluid.personalData");
+fluid.personalData.startServerCmd = "node startServer.js";
+
+const NUM_CHECK_REQUESTS = 10;
+const DELAY = 2000; // msec
+
+fluid.personalData.sleep = async function (delay) {
+    return new Promise((resolve) => setTimeout(resolve, delay));
+};
+
+/**
+ * Check if the personal data server is running using the given url.  If not
+ * running, start it. If it does not start after ten attempts, give up.
+ *
+ * @param {Number} serverPort - The port that the server is listening to.
+ * @return {Object} An object that has the status of the last request regardless
+ *                  of success, and the node ChildProcess started by `exec()`
+ *                  or `null` if it failed to start.
+ */
+fluid.personalData.startServer = async function (serverPort) {
+    const url = "http://localhost:" + serverPort;
+    let resp = {};
+
+    console.debug(`- Checking if server is running using ${url}`);
+    try {
+        resp = await axios(url);
+        return { status: resp.status, process: null, wasRunning: true };
+    }
+    catch (error) { ; }
+
+    // Server needs to be started
+    console.debug(`- Starting server at ${url}`);
+    const pdServerProcess = exec(fluid.personalData.startServerCmd + " " + serverPort);
```

Comment:
It might be less fragile, and also enable greater use of "white-box testing" in future, to start this server in the same process by a direct function call